File: Non-decreasing Array/non-decreasing-array.py
from typing import List


def checkPossibility(nums: List[int]) -> bool:
    pos = 0
    for i in range(1, len(nums)):
        if nums[i - 1] > nums[i]:
            pos += 1
            if (i - 2) > -1 and nums[i - 2] > nums[i]:
                nums[i] = nums[i - 1]
            else:
                nums[i - 1] = nums[i]

    return pos < 2


print(checkPossibility([4, 2, 3]))  # True
print(checkPossibility([4, 2, 1]))  # False

File: Non-decreasing Array/non-decreasing-array.php
<?php
declare(strict_types=1);

/**
 * Created by PhpStorm.
 * User: kovalevcon
 * Date: 07.06.19
 * Time: 17:51
 */

/**
 * Check if it could become non-decreasing by modifying at most 1 element in array of integers
 *
 * @param Integer[] $nums
 * @return Boolean
 */
function checkPossibility(array $nums): bool
{
    $pos = null;
    $length = count($nums);
    for ($i = 0; $i < $length - 1; $i++) {
        if ($nums[$i] > $nums[$i + 1]) {
            if (!is_null($pos)) {
                return false;
            }
            $pos = $i;
        }
    }

    return (is_null($pos) || $pos === 0 || $pos === $length - 2 ||
        $nums[$pos - 1] <= $nums[$pos + 1] || $nums[$pos] <= $nums[$pos + 2]);
}

echo "<pre>checkPossibility([4, 2, 3])\r\n";
var_dump(checkPossibility([4, 2, 3])); // true
echo "</pre>";

echo "<pre>checkPossibility([4, 2, 1])\r\n";
var_dump(checkPossibility([4, 2, 1])); // true
echo "</pre>";